src/platform/packages/shared/kbn-rrule/sanitize.ts

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the "Elastic License
+ * 2.0", the "GNU Affero General Public License v3.0 only", and the "Server Side
+ * Public License v 1"; you may not use this file except in compliance with, at
+ * your election, the "Elastic License 2.0", the "GNU Affero General Public
+ * License v3.0 only", or the "Server Side Public License, v 1".
+ */
+
+import type { Options } from './types';
+
+export function sanitizeOptions(opts: Options) {
+  const options = { ...opts };
+
+  // Guard against invalid options that can't be omitted
+  if (!options.dtstart) {
+    throw new Error('Cannot create RRule: dtstart is required');
+  }
+
+  if (!options.tzid) {
+    throw new Error('Cannot create RRule: tzid is required');
+  }
+
+  if (isNaN(options.dtstart.getTime())) {
+    throw new Error('Cannot create RRule: dtstart is an invalid date');
+  }
+
+  if (options.until && isNaN(options.until.getTime())) {
+    throw new Error('Cannot create RRule: until is an invalid date');
+  }
+
+  if (typeof options.interval !== 'number') {
```

Comment:
I added this additional validation lmk what you think

Reply:
Should we make it `options.interval && typeof options.interval !== 'number'`? As interval is optional param.

Reply:
i did it, lets hope it fixes the billion failing tests my commit caused